greeter/README.md

```diff
@@ -1,21 +1,510 @@
+We'd like to start the caching service during the start of the application.  Likewise, we'd also want to stop the caching service when the application stops.  To do this, we are going to use a [lifecycle observer](https://loopback.io/doc/en/lb4/Life-cycle.html).  
+
+
+1. Run the observer generator `lb4 observer` command.
+    ```sh
+    $ lb4 observer
+    ? Observer name: Cache
+    ? Observer group: 
+    create src/observers/cache.observer.ts
+    update src/observers/index.ts
+
+    Observer Cache was created in src/observers/
+    ```
+
+2. Go to `src/observers/cache.observer.ts`, modify the constructor to get the caching service: 
+
+    ```ts
+    constructor(
+        @inject(CACHING_SERVICE) private cachingService: CachingService,
+    ) {}
+    ```
+
+    And the following import statements:
+
+    ```ts
+    import {inject} from '@loopback/context';
+    import {CachingService} from '../services/caching.service';
+    import {CACHING_SERVICE} from '../keys';
+    ```
+
+3. For the `start()` and `stop()` function, we want to start and stop the caching service.
+
+    ```ts
+    /**
+     * This method will be invoked when the application starts
+     */
+    async start(): Promise<void> {
+        // Add your logic for start
+        await this.cachingService.start();
+    }
+
+    /**
+     * This method will be invoked when the application stops
+     */
+    async stop(): Promise<void> {
+        // Add your logic for stop
+        await this.cachingService.stop();
+    }
+    ```
+
+
+### Step 3: Create global interceptor for caching
+
+1. Run the interceptor generator `lb4 interceptor` command.
+
+    ```sh
+    $ lb4 interceptor
+    ? Interceptor name: caching
+    ? Is it a global interceptor? Yes
+
+    Global interceptors are sorted by the order of an array of group names bound to ContextBindings.GLOBAL_INTERCEPTOR_ORDERED_GROUPS. See https://loopback.io/doc/en/lb4/Interceptors.html#order-of-invocation-for-interceptors.
+
+    ? Group name for the global interceptor: ('') 
+    create src/interceptors/caching.interceptor.ts
+    update src/interceptors/index.ts
+
+    Interceptor Caching was created in src/interceptors/
+    ```
+
+2. In `src/interceptors/caching.interceptors.ts`, modify the constructor:
+
+    ```ts
+    constructor(
+        @inject(CACHING_SERVICE) private cachingService: CachingService,
+    ) {}
+    ```
+
+    Add and modify the following import statements:
+
+    ```ts
+    import {
+        /* inject, */
+        globalInterceptor,
+        Interceptor,
+        InvocationContext,
+        InvocationResult,
+        Provider,
+        inject,
+        ValueOrPromise,
+    } from '@loopback/context';
+    import {CachingService} from '../services/caching.service';
+    import {CACHING_SERVICE} from '../keys';
+    import {RestBindings} from '@loopback/rest';
+    ```
+
+3. Add the pre-invocation and post-invocation logics in the `intercept` function.
+
+    ```ts
+    // Add pre-invocation logic here
```

Comment:
we need to exclude two endpoints for the interceptor:
```ts
const EXCLUDED_PATHS = ['/explorer/', '/explorer/openapi.json'];
// And then skip them in the intercept function
```
Otherwise when hit the explorer endpoint, it's response also get cached but no timestamp in it, which breaks the cache interceptor.

Reply:
I hit this bug. See comment below.